### PR TITLE
Add support for doing uploads in POST.

### PR DESCRIPTION
**Pull request by amouat**

### CODE REVIEW

File: src/routes.rs [hunk 1]
```diff
@@ -454,41 +457,84 @@ fn post_blob_upload(
...
+                        }
+                        Err(_) => Err(Error::InternalError),
+                    }
+                }
+                Err(_) => {
+                    // TODO: this conflates rpc errors with uuid not existing
+                    // TODO: pipe breaks if we don't accept the whole file
+                    // Possibly makes us prone to DOS attack?
+                    warn!("Uuid {} does not exist, piping to /dev/null", &up_info.uuid());
+                    let _ = data.stream_to_file("/dev/null");
```
Comment (mcwienczek) [inline src/routes.rs]: can't we just break the connection instead?

Comment (amouat) [reply src/routes.rs]: Not sure, I've not really looked into this tbh

Comment (amouat) [reply src/routes.rs]: I think this is best left for another PR - do you want to investigate?

Comment (mcwienczek) [reply src/routes.rs]: I will create an issue for it, meanwhile I will approve

File: src/routes.rs [hunk 2]
```diff
@@ -454,41 +457,84 @@ fn post_blob_upload(
...
+                            rt.block_on(r).map_err(|_| Error::InternalError).map(|au| Upload::Accepted(au))
+                        }
+                        Err(_) => Err(Error::InternalError),
+                    }
+                }
+                Err(_) => {
+                    // TODO: this conflates rpc errors with uuid not existing
+                    // TODO: pipe breaks if we don't accept the whole file
+                    // Possibly makes us prone to DOS attack?
+                    warn!("Uuid {} does not exist, piping to /dev/null", &up_info.uuid());
```
Comment (mcwienczek) [inline src/routes.rs]: #55

**Pull request merged**